### PR TITLE
Add user impersonation information to `UserHeaderItem`

### PR DESCRIPTION
**Pull request by fichtnerma**
## Description

Update UserHeaderItem  to include Impersonation Inlay

## Example

-   [x] I have verified if my change requires an example

## Screenshots/screencasts
image

image

## Changeset

-   [x] I have verified if my change requires a changeset

## Related tasks and documents
https://vivid-planet.atlassian.net/browse/COM-1125
https://github.com/vivid-planet/comet/pull/2698/files#diff-06d06a04613e0ce6411966a8b4987819ddafdda7838b0d120422de246051c696

## Open TODOs/questions

-   [x] Merge https://github.com/vivid-planet/comet/pull/2698/files#diff-06d06a04613e0ce6411966a8b4987819ddafdda7838b0d120422de246051c696

### CODE REVIEW

File: packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx [hunk 1]
```diff
@@ -40,62 +47,73 @@ interface UserHeaderItemProps {
...
-                >
-                    <FormattedMessage id="comet.about" defaultMessage="About" />
-                </Button>
-                {children}
+        <AppHeaderDropdown
+            buttonChildren={buttonChildren ?? (isMobile ? <UserHeaderAccountIcon impersonated={user.impersonated} /> : user.name)}
+            startIcon={isMobile ? undefined : <UserHeaderAccountIcon impersonated={user.impersonated} />}
+        >
+            <DropdownContent padding={0}>
+                <Box padding={4}>
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx:63]: Please use either styled components or Box directly, but not both. @jamesricky what do you prefer?

File: packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+const Tag = styled("div")`
+    display: flex;
+    align-items: center;
+    font-size: 12px;
+    padding: 4px 10px;
+    line-height: 16px;
+    justify-content: center;
+    background-color: ${({ theme }) => theme.palette.primary.main};
+    border-radius: 12px;
+`;
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx:37]: Can't we use a normal Chip component for this?

File: packages/admin/cms-admin/src/common/header/impersonation/UserHeaderAccountIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                    <Account color="inherit" />
+                </ImpersonatedIcon>
+            )}
+        </AccountIconWrapper>
+    );
+}
+
+const AccountIcon = styled("div", {
+    shouldForwardProp: (prop) => prop !== "active",
+})<{ active?: boolean }>`
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/header/impersonation/UserHeaderAccountIcon.tsx:25]: Should we use AvatarGroup for this? https://mui.com/material-ui/react-avatar/#grouped

File: packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            <Tag>
+                <FormattedMessage id="comet.impersonation.inlay.title" defaultMessage="Impersonation mode" />
+            </Tag>
+            <Typography variant="h6" sx={{ marginTop: 3 }}>
+                {user.name}
+            </Typography>
+            <Typography variant="body2" sx={{ marginBottom: 3 }}>
+                {user.email}
+            </Typography>
+            <StopImpersonationButton startIcon={<Reset />} fullWidth variant="contained" color="secondary" sx={{ justifyContent: "center" }} />
```
Comment (johnnyomair) [inline packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx:23]: I believe the text of this button needs to be updated. It's currently "Stop Impersonation"

File: packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx (hunk 1 above)
Comment (jamesricky) [reply packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx:63]: IMO `Box` is fine if it's just one or two styling properties, if there is more styling I'd prefer a styled component.

File: packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx [hunk 5]
```diff
@@ -40,62 +47,73 @@ interface UserHeaderItemProps {
...
+                            onClick={() => {
+                                setShowAboutModal(true);
+                            }}
+                            color="primary"
+                            sx={{ padding: 0 }}
+                        >
+                            <Typography variant="caption" fontWeight={300}>
+                                <FormattedMessage id="comet.about" defaultMessage="About/Copyright" />
+                            </Typography>
+                        </Button>
```
Comment (jamesricky) [inline packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx:114]: Custom button/text styling should not be needed

File: packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx [hunk 6]
```diff
@@ -40,62 +47,73 @@ interface UserHeaderItemProps {
...
+                            variant="contained"
+                            color="primary"
+                            startIcon={<Logout />}
+                            onClick={async () => {
+                                const result = await signOut();
+                                if (result.data) {
+                                    location.href = result.data.currentUserSignOut;
+                                }
+                            }}
+                            sx={{ justifyContent: "center" }}
```
Comment (jamesricky) [inline packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx:95]: Custom button styling should not be needed

File: packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx (hunk 4 above)
Comment (jamesricky) [inline packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx:23]: Custom button styling should not be needed

**Review by jamesricky — COMMENTED** [review]
There are some inconsistencies compared to the design (text styling, button size, missing border)

| Design | Implementation |
| --- | --- |
| Design | Implementation |

File: packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx (hunk 6 above)
Comment (fichtnerma) [reply packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx:95]: Without this the button text won t be centered

File: packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx (hunk 4 above)
Comment (fichtnerma) [reply packages/admin/cms-admin/src/common/header/impersonation/ImpersonationInlay.tsx:23]: variant, color and fullWidth is required

**Comment by fichtnerma** [comment]
> There are some inconsistencies compared to the design (text styling, button size, missing border)
> 
> Design	Implementation
> Design	Implementation

The sizes are now the same as the Figma Component

File: packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx (hunk 6 above)
Comment (jamesricky) [reply packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx:97]: The default styling should be sufficient. 
Is this the issue? Can this be removed or just not used here? 

https://github.com/vivid-planet/comet/blob/bf8905bd4004e90c0643ce3d865982be1d31ab5e/packages/admin/cms-admin/src/common/header/UserHeaderItem.tsx#L20-L22

**Comment by johnnyomair** [comment]
I will review this again once #2698 has been merged.

File: .changeset/eight-owls-shop.md
Comment (johnnyomair) [inline .changeset/eight-owls-shop.md:1]: This is from the parent PR, please rebase to remove commits and files that have already been merged

File: packages/admin/admin/src/messages.d.ts
Comment (johnnyomair) [inline packages/admin/admin/src/messages.d.ts:1]: This seems like a mistake, please remove.

File: packages/admin/admin/src/messages.d.ts.map
Comment (johnnyomair) [inline packages/admin/admin/src/messages.d.ts.map:1]: Please remove this as well.

**Review by johnnyomair — CHANGES_REQUESTED** [review]

**Pull request merged**